### ISSUE TITLE
Add a `date_range` parameter to the WebSearch tool

### ISSUE THREAD

**Issue by shhlife** [griptape-ai/griptape#823]
- [x ] I have read and agree to the [contributing guidelines](https://github.com/griptape-ai/griptape#contributing).

**Is your feature request related to a problem? Please describe.**
One of our customers would like the ability to specify a date_range within web search. I provided an example how to do it with rules, and how to modify the WebSearch tool to do it - but it would be nice to have it as part of the framework.

**Describe the solution you'd like**
An optional parameter to the WebSearch tool to include a date range. 

The Google search api requires it to be in the format:
`&sort=date:r:YYYYMMDD:YYYYMMDD`

For example:
`&sort=date:20240101:20240401`

To add it to the tool, I added an optional parameter for date_range:
```
                    schema.Optional(Literal(
                        "date_range",
                        description="Optional date range to search within. Format: YYYYMMDD:YYYYMMDD",
                    )): str

```

And then added some logic to the `search` method to use it:

```
    def search(self, props: dict) -> ListArtifact | ErrorArtifact:
        query = props["values"]["query"]
        date_range = props["values"].get("date_range", None)
        if date_range:
            query +=f"&sort=date:r:{date_range}"
        try:
            return ListArtifact([TextArtifact(str(result)) for result in self._search_google(query)])
        except Exception as e:
            return ErrorArtifact(f"error searching Google: {e}")
```

**Describe alternatives you've considered**
Tried adding it as a ruleset as well - that worked but didn't feel as solid.

```
agent = Agent(
    rulesets=[search_ruleset],
    rules=[
        Rule(
            "When asked to search within a date range, use the `date_range` parameter of the Search Client. Example: 20210101:20220101"
        )
    ],
    tools=[search_client],
)
```

**Comment by shhlife** [comment]
With the latest update in 0.28.0, is it possible to take a look at this one again? Would it need to be on the driver?

**Comment by collindutter** [comment]
Maybe -- do you want the LLM to come up with the date? Or do you want to set it yourself?

**Comment by collindutter** [comment]
Closed in #1015
